source/app/service-providers/citeproc/index.ts

```diff
@@ -95,11 +99,22 @@ export default class CiteprocProvider {
    */
   private readonly _sys: CSLKernel
 
-  constructor () {
-    global.log.verbose('Citeproc provider booting up ...')
+  private readonly _logger: LogProvider
+  private readonly _config: ConfigProvider
+  private readonly _notifications: NotificationProvider
+  private readonly _windows: WindowProvider
+
+  constructor (logger: LogProvider, config: ConfigProvider, notifications: NotificationProvider, windows: WindowProvider) {
+    super()
+    this._logger = logger
+    this._config = config
+    this._notifications = notifications
+    this._windows = windows
+
+    this._logger.verbose('Citeproc provider booting up ...')
```

Comment:
What is the distinction between the constructor and the boot methods?

Reply:
In general, I thought of that like so: The constructors should make sure that the objects can be accessed, but any "impure" code (i.e. disk access etc.) is async and therefore should be part of the boot methods. This also has the benefit that most stuff that's running doesn't run into race conditions anymore.